agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/NetworkStatsbeatHttpPipelinePolicy.java

```diff
@@ -57,17 +57,21 @@ public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineN
               } else if (StatusCodes.isRedirect(statusCode)) {
                 // these are not tracked as success or failure since they are just redirects
               } else if (statusCode == 402 || statusCode == 439) {
-                networkStatsbeat.incrementThrottlingCount(instrumentationKey, host);
+                networkStatsbeat.incrementThrottlingCount(
+                    instrumentationKey, host, Integer.valueOf(statusCode));
               } else if (StatusCodes.isRetryable(statusCode)) {
-                networkStatsbeat.incrementRetryCount(instrumentationKey, host);
+                networkStatsbeat.incrementRetryCount(
+                    instrumentationKey, host, Integer.valueOf(statusCode));
               } else {
-                // note: 401 and 403 are currently tracked as failures
-                networkStatsbeat.incrementRequestFailureCount(instrumentationKey, host);
+                // 400 and 404 will be tracked as failure count
+                networkStatsbeat.incrementRequestFailureCount(
+                    instrumentationKey, host, Integer.valueOf(statusCode));
               }
             })
         .doOnError(
             throwable -> {
-              networkStatsbeat.incrementExceptionCount(instrumentationKey, host);
+              networkStatsbeat.incrementExceptionCount(
+                  instrumentationKey, host, throwable.getClass().toString());
```

Comment:
`getName` is more common

```suggestion
                  instrumentationKey, host, throwable.getClass().getName());
```